### ISSUE TITLE
HTML Anchor instructions are unclear

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15306]
Instructions are unclear

-   **Severity:** Low
-   **Affected Populations:**
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Block Panel

#### Issue description

Users going through the Block panel after selecting a Heading type
block, will come to a field asking for an "HTML anchor", with the
explanation: "Anchors lets you link directly to a section on a page."

While it may be too technical to explain that this field adds an `id`
attribute to the heading, the brief explanation leads users to be unsure
if the field wants an actual anchor tag or a string, and what the rules
are for its value.

Users with cognitive, reading or writing disabilities may also have
trouble with the input's automatic changing of space and other
characters as they type, without explanation.

##### Issue Code

```
    <div class="components-base-control__field">


        <label class="..." for="inspector-text-control-4">HTML Anchor</label>


        <input class="..." type="text" id="inspector-text-control-4" aria-describedby="inspector-text-control-4__help" value="...">


    </div>


    <p id="inspector-text-control-4__help" class="...">Anchors lets you link directly to a section on a page.</p>
```

#### Remediation Guidance

Change "anchor" to something that doesn't suggest the user should
actually add an anchor. This also avoids web-developer jargon: more
users know what a "link" is than an "anchor".

Have the explanatory text mention the sole restriction of id-token
strings (i.e. they cannot contain spaces) to help explain why spaces are
being automatically changed to other characters.

Consider showing an example of resulting HTML code so that users can see
what changes as they change their string. Users who don't understand
HTML code are still free to ignore this.

##### Recommended Code

```
    <div class="components-base-control__field">


        <label class="..." for="inspector-text-control-4">Add an-page link</label>


        <input class="..." type="text" id="inspector-text-control-4" aria-describedby="inspector-text-control-4__help" value="...">


    </div>


    <p id="inspector-text-control-4__help" class="...">Add a unique name (without spaces) to your heading to allow linking directly to that section on the page.</p>





    <div class="...">


        <label class="..." for="inspector-text-4__help">Example code result</label>


        <input class="..." type="text" id="inspector-text-4__help" readonly="" value="[heading with id here]">


    </div>
```

#### Relevant standards

-   [3.3.2 Labels or Instructions
    (Level A)](https://www.w3.org/TR/WCAG20/#minimize-error-cues)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-39 in Tenon's report

**Comment by melchoyce** [comment]
Full report screenshot:

![image](https://user-images.githubusercontent.com/2846578/57186940-73e74280-6e9c-11e9-94bf-dff253ad84e4.png)

**Comment by afercia** [comment]
+1 Some clearer copy may help.

**Comment by melchoyce** [comment]
@michelleweber is trying to reply to this thread, but for some reason is  getting an error? She sent me this via DM:

> > Change "anchor" to something that doesn't suggest the user should actually add an anchor. This also avoids web-developer jargon: more users know what a "link" is than an "anchor".
> 
> Since the name for this section is "HTML Anchor," I read the explanatory text as defining that term; I think we can add more clarity, but if the name of this item is HTML Anchor, we can't/shouldn't try to get away from that entirely. Or do you want to change both the name and the explanation? It wasn't clear to me. Also, this seems like a good place for a "learn more" link to info on anchors.
> 
> For the explanatory text, maybe something like:
>
> "Enter a word or two — no spaces, please! — to make a unique web address just for this heading, called an "anchor." Then, you'll be able link directly to this section of your page. Learn more about anchors."
> 
> (I think this can be better, but want to make sure I'm on the right track first.)

**Comment by noisysocks** [comment]
#16142 fixed this.
